deps/wazuh_testing/wazuh_testing/process_resource_monitoring/src/wazuh_metrics.py

```diff
@@ -0,0 +1,334 @@
+# Copyright (C) 2015-2024, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+"""Script to monitor Wazuh central components resource usage."""
+
+
+import argparse
+import logging
+from collections import defaultdict
+from datetime import datetime
+from os import makedirs
+from os.path import join
+from signal import SIGINT, SIGTERM, signal
+from tempfile import gettempdir
+from time import sleep, time_ns
+from types import FrameType
+from typing import Dict, List, Optional
+
+from process_resource_monitoring._logger import logger
+from process_resource_monitoring.disk_usage_tracker import DiskUsageTracker
+from process_resource_monitoring.monitor import Monitor
+
+# Constant variables
+METRICS_FOLDER: str = join(gettempdir(), 'process_metrics')
+CURRENT_SESSION: str = join(METRICS_FOLDER, datetime.now().strftime('%d-%m-%Y'), str(time_ns() // 10**9))
+PROCESS_MAPPING: Dict[str, str] = {
+    # Manager processes
+    'agentlessd': 'wazuh-agentlessd',
+    'analysisd': 'wazuh-analysisd',
+    'apid': 'wazuh_apid.py',
+    'authd': 'wazuh-authd',
+    'clusterd': 'wazuh_clusterd.py',
+    'csyslogd': 'wazuh-csyslogd',
+    'db': 'wazuh-db',
+    'dbd': 'wazuh-dbd',
+    'execd': 'wazuh-execd',
+    'integratord': 'wazuh-integratord',
+    'logcollector': 'wazuh-logcollector',
+    'maild': 'wazuh-maild',
+    'modulesd': 'wazuh-modulesd',
+    'monitord': 'wazuh-monitord',
+    'remoted': 'wazuh-remoted',
+    'wazuh-syscheckd': 'syscheckd',
+
+    # Indexer
+    'indexer': 'wazuh-indexer',
+
+    # Dashboard
+    'dashboard': 'wazuh-dashboard',
+
+    # Agent
+    'agentd': 'wazuh-agentd',
+}
```

Comment:
Why enforce the processes?

Reply:
Following the rationale of https://github.com/wazuh/wazuh-qa/pull/5747#discussion_r1773569369 chances are it will be "better" (less time, resources) to simply take a "naive approach" and modify it whenever it is needed. To address differences between versions, each branch/tag would have the script with the correct parameters. There are other ways of handling so, for example, having a mapping with all the relevant processes across versions and enforce the one specific to only one version.

Reply:
Resolved in a meeting with @jseg380